### PR TITLE
Adds an option to allow vagrant to run the provisioning step

### PR DESCRIPTION
**Pull request by RuriRyan**
There's a new module argument called "no_provision" which, if set
to False, causes Vagrant to run the provisioning step.

I ran into a problem where a Vagrant Box wouldn't work because it has a Vagrantfile included with some shell script, which is needed for the Box to work correctly.

### CODE REVIEW

File: test/scenarios/driver/vagrant/molecule/default/tests/test_default.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 
 def test_hostname(host):
-    assert 'instance' == host.check_output('hostname -s')
+    assert host.check_output('hostname -s') == 'instance'
```
Comment (retr0h) [inline test/scenarios/driver/vagrant/molecule/default/tests/test_default.py]: Why is this flipped?  Looks unrelated and our pattern is `expected == returned`.

Comment (RuriRyan) [reply test/scenarios/driver/vagrant/molecule/default/tests/test_default.py]: This was actually the only test case which was written as `expected == returned` and my linter complained that test cases should actually be `returned == expected`. But I can revert it back, if it bothers you.

**Comment by madrover** [comment]
Looking forward to this one!

**Comment by retr0h** [comment]
We need to upgrade the cookiecutter templates.

* https://github.com/metacloud/molecule/blob/master/molecule/cookiecutter/scenario/driver/vagrant/%7B%7Bcookiecutter.molecule_directory%7D%7D/%7B%7Bcookiecutter.scenario_name%7D%7D/create.yml
* https://github.com/metacloud/molecule/blob/master/molecule/cookiecutter/molecule/%7B%7Bcookiecutter.role_name%7D%7D/%7B%7Bcookiecutter.molecule_directory%7D%7D/%7B%7Bcookiecutter.scenario_name%7D%7D/molecule.yml#L36-L39

**Comment by nisimond** [comment]
I am, as a matter of principle, opposed to a value of `False` being used to mean "do something".

If you flip the semantics on this option, it will work the same, and IMO be clearer.

**Comment by retr0h** [comment]
@RuriRyan please update with feedback provided.  Thanks.

**Comment by retr0h** [comment]
Closing have not heard back from submitter.  Please reopen when you are able and willing.  Thanks again.
